### PR TITLE
Convert Fix Goldens to dart script

### PR DESCRIPTION
**Pull request by CoderDake**

### CODE REVIEW

File: tool/lib/commands/fix_goldens.dart [hunk 1]
```diff
@@ -0,0 +1,101 @@
+// #!/bin/bash -e
```
Comment (kenzieschmoll) [inline tool/lib/commands/fix_goldens.dart:1]: can we get rid of these? also add copyright please

File: tool/lib/commands/fix_goldens.dart [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+import 'dart:async';
+import 'dart:io';
+
+import 'package:args/command_runner.dart';
+import 'package:io/io.dart';
+import 'package:path/path.dart' as path;
+
+import '../utils.dart';
+
+class FixGoldensCommand extends Command {
```
Comment (kenzieschmoll) [inline tool/lib/commands/fix_goldens.dart:12]: dart doc

File: tool/lib/commands/fix_goldens.dart [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+
+  @override
+  FutureOr? run() async {
+    // Change the CWD to the repo root
+    Directory.current = pathFromRepoRoot("");
+
+    final runId = argResults!['run-id']!;
+    final tmpDownloadDir = Directory(
+      path.join('.tmp${DateTime.now().millisecondsSinceEpoch}'),
+    );
```
Comment (kenzieschmoll) [inline tool/lib/commands/fix_goldens.dart:39]: can we use `Directory.systemTemp.createTemp` instead?

File: tool/lib/commands/fix_goldens.dart [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+      await processManager.runProcess(
+        CliCommand.from('gh', [
+          'run',
+          'download',
+          runId,
+          '-p',
+          '*golden_image_failures*',
+          '-D',
+          tmpDownloadDir.path,
+        ]),
```
Comment (kenzieschmoll) [inline tool/lib/commands/fix_goldens.dart:52]: trailing comma. We should make sure the trailing commas lint is applied to the tool/ directory

File: tool/lib/commands/fix_goldens.dart (hunk 1 above)
Comment (CoderDake) [reply tool/lib/commands/fix_goldens.dart:1]: yep this was accidentally left behind

File: tool/lib/commands/fix_goldens.dart [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+        }
+        await downloadedGolden.rename(destinationPath);
+
+        print("Fixed: $destinationPath");
+      }
+
+      print('Done updating ${downloadedGoldens.length} goldens');
+    } finally {
+      tmpDownloadDir.deleteSync(recursive: true);
+    }
```
Comment (kenzieschmoll) [inline tool/lib/commands/fix_goldens.dart:99]: I think you need to call

```
    // Closes stdin for the entire program.
    await sharedStdIn.terminate();
```
at the end or this script will not terminate due to an open port

File: tool/lib/commands/fix_goldens.dart (hunk 4 above)
Comment (CoderDake) [reply tool/lib/commands/fix_goldens.dart:52]: @kenzieschmoll which lint is this? Currently tool has `require_trailing_commas`

File: tool/lib/commands/fix_goldens.dart (hunk 5 above)
Comment (CoderDake) [reply tool/lib/commands/fix_goldens.dart:104]: I haven't had any issues with this yet. What makes you think the stdin would hang?

**Review by kenzieschmoll — APPROVED** [review]

**Pull request merged**